src/org/mozilla/javascript/Kit.java

```diff
@@ -56,11 +97,23 @@
      */
     public static Class<?> classOrNull(ClassLoader loader, String className)
     {
+
+        LRUCache lruCache = null;
+        if (enabledNonClassCaching) {
+            lruCache = getOrCreateCache(loader);
+
+            if (lruCache.containsKey(className))
```

Comment:
I don't know that we can assume that LRUcache is only used by one thread, so I think this needs to be synchronized.